New Module for Solax Gen5

rgrae81 opened the pull request:
Added new Module vor Solax GEN5 Inverters with up to 3 PV Strings and up to 2 Batteries

Adresses Issue [1919](https://github.com/openWB/core/issues/1919)

UI: [openwb-ui-settings/pull/570](https://github.com/openWB/openwb-ui-settings/pull/570)

rgrae81 committed:
New Module for Solax Gen5

Added new Module vor Solax GEN5 Inverters with up to 3 PV Strings and up to 2 Batteries

rgrae81 committed:
Merge branch 'openWB:master' into master

rgrae81 committed:
Update inverter.py

Error: packages/modules/devices/solax/solax_gen5/inverter.py:3:1: F401 'pymodbus.constants.Endian' imported but unused

rgrae81 committed:
Update bat.py

Error: packages/modules/devices/solax/solax_gen5/bat.py:36:15: E111 indentation is not a multiple of 4
Error: packages/modules/devices/solax/solax_gen5/bat.py:38:15: E111 indentation is not a multiple of 4
Error: packages/modules/devices/solax/solax_gen5/bat.py:41:15: E111 indentation is not a multiple of 4
Error: packages/modules/devices/solax/solax_gen5/bat.py:42:15: E111 indentation is not a multiple of 4
Error: packages/modules/devices/solax/solax_gen5/bat.py:44:15: E111 indentation is not a multiple of 4
Error: packages/modules/devices/solax/solax_gen5/bat.py:54:1: E305 expected 2 blank lines after class or function definition, found 1

rgrae81 committed:
Update counter.py

Error: packages/modules/devices/solax/solax_gen5/counter.py:31:121: E501 line too long (138 > 120 characters)
Error: packages/modules/devices/solax/solax_gen5/counter.py:46:29: E122 continuation line missing indentation or outdented
Error: packages/modules/devices/solax/solax_gen5/counter.py:50:22: E124 closing bracket does not match visual indentation
Warning: packages/modules/devices/solax/solax_gen5/counter.py:51:1: W293 blank line contains whitespace

rgrae81 committed:
Update device.py

 Error: packages/modules/devices/solax/solax_gen5/device.py:10:121: E501 line too long (127 > 120 characters)

LKuemmel commented on packages/modules/devices/solax/solax_gen5/inverter.py:
```diff
@@ -0,0 +1,43 @@
+            power_pv3 = self.__tcp_client.read_input_registers(292, ModbusDataType.UINT_16, unit=self.__modbus_id) * -1
+            power_temp = (power_pv1, power_pv2, power_pv3)
+            power = sum(power_temp)
+            exported = self.__tcp_client.read_input_registers(80, ModbusDataType.UINT_16, unit=self.__modbus_id) * 100
```
Ist beim Solax Gen5 Register 80 der Gesamt-Ertrag oder der Tagesertrag?

rgrae81 replied:
Tagesertrag
![grafik](https://github.com/user-attachments/assets/5162a407-4cac-4fbb-9d06-d80af9b3754d)

LKuemmel replied:
Das ist der Gesamtertrag.

benderl replied:
Von der Regelung wird aktuell bei den WR und Speichern nur die Leistung verwendet. Du kannst aber z. B. die Ströme gerne mit auslesen. Die müssen auch nicht 1:1 zur Wirkleistung passen.

rgrae81 replied:
Dann lass ich die Ströme mit drin.

ndrsnhs replied:
Viele Register stimmen mit dem Ursprungsmodul überein. Es liegen hier nur vereinzelte Registerabweichungen vor (bspw zusätzliches Erfassen von PV3 Power; SoC Register 302 statt 28; Auslesen der Gesamtzählerstände).
Es wäre besser das als zusätzliche Option zur Auswahl im bestehenden Modul anzubieten statt ein eigenes Modul zu erstellen (auch um redundanten Code zu vermeiden).

rgrae81 replied:
@ndrsnhs Ist auch ein Ansatz. Wo kann man denn die UI für die Einstellungsseite in einem laufenden System zum testen anpassen?

ndrsnhs replied:
@rgrae81 
https://github.com/openWB/core/wiki/Einstellungs-Seite-erstellen
Das settings-Repo muss geforkt werden und dann die dortigen Änderungen kompilieren.

rgrae81 committed:
Update inverter.py

Changed exported from daily register to total register

rgrae81 committed:
Update bat.py

Error: packages/modules/devices/solax/solax_gen5/bat.py:41:121: E501 line too long (122 > 120 characters)
Error: packages/modules/devices/solax/solax_gen5/bat.py:42:121: E501 line too long (122 > 120 characters)
Error: packages/modules/devices/solax/solax_gen5/bat.py:54:1: E305 expected 2 blank lines after class or function definition, found 1
Warning: packages/modules/devices/solax/solax_gen5/bat.py:55:1: W391 blank line at end of file

rgrae81 committed:
Merge branch 'openWB:master' into master

kevinwieland commented:
@rgrae81 wie ist hier der Stand?
